src/languages/golfscript/index.ts

```diff
@@ -0,0 +1,92 @@
+import { functionCall, id, rangeIndexCall } from "../../IR";
+import { defaultDetokenizer, Language } from "../../common/Language";
+
+import emitProgram from "./emit";
+import {
+  mapOps,
+  mapPrecedenceOps,
+  plus1,
+  useIndexCalls,
+} from "../../plugins/ops";
+import { renameIdents } from "../../plugins/idents";
+import { evalStaticExpr } from "../../plugins/static";
+import { flipBinaryOps } from "../../plugins/binaryOps";
+import { golfLastPrint } from "../../plugins/print";
+import { addImports } from "./plugins";
+
+const golfscriptLanguage: Language = {
+  name: "Golfscript",
+  extension: "txt", // Golfscript doesn't appear to have a custom extension
+  emitter: emitProgram,
+  golfPlugins: [flipBinaryOps, evalStaticExpr, golfLastPrint()],
+  emitPlugins: [useIndexCalls()],
+  finalEmitPlugins: [
+    mapOps([
+      ["true", (_) => id("1", true)],
+      ["false", (_) => id("0", true)],
+      ["println", (x) => functionCall(x, "puts")],
+      ["print", (x) => functionCall(x, "print")],
```

Comment:
Does the current system make it feasible to use `p` for printing integers?

Reply:
It does, yes. In the current system, `println` and `print` can be feasibly implemented as "n" and "", though this likely isn't a very future-proofed approach.